### PR TITLE
Create README.md for examples/

### PR DESCRIPTION
**Pull request by sbhavani**
# Description

Our examples are split between examples/ and docs/examples/. We also have features (e.g. inference) hidden in our examples that would be worth summarizing in a single README. 

I'd also like to extend this to include third party examples (e.g. Hugging Face Accelerate) using TE.

## Type of change

- [x] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

Please list the changes introduced in this PR:

- Created a new README.md

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - **FP8 Weight Caching**: Avoiding redundant FP8 casting during multiple gradient accumulation steps to improve efficiency.
+- [Introduction to FP8](https://github.com/NVIDIA/TransformerEngine/blob/main/docs/examples/fp8_primer.ipynb)
+  - Overview of FP8 datatypes (E4M3, E5M2), mixed precision training, delayed scaling strategies, and code examples for FP8 configuration and usage.
+- [TE Quickstart](https://github.com/NVIDIA/TransformerEngine/blob/main/docs/examples/quickstart.ipynb)
+  - Introduction to TE, building a Transformer Layer using PyTorch, and instructions on integrating TE modules like Linear and LayerNorm.
+- [Basic MNIST Example](https://github.com/NVIDIA/TransformerEngine/tree/main/examples/pytorch/mnist)
+
+# JAX
+
+# PaddlePaddle
```
Comment (ksivaman) [inline examples/README.md:34]: We should either fill or remove these

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Examples
+
+We provide a variety of examples for deep learning frameworks including [PyTorch](https://github.com/huggingface/transformers/tree/main/examples/pytorch), [JAX](https://jax.readthedocs.io/en/latest/), and [PaddlePaddle](https://github.com/PaddlePaddle/Paddle).
```
Comment (ksivaman) [inline examples/README.md:3]: The 3 framework links all lead to different things:
1. PyTorch leads to HF PyTorch examples.
2. Jax leads to its docs
3. And PaddlePaddle leads to the code.
Let's make it consistent.

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Examples
+
+We provide a variety of examples for deep learning frameworks including [PyTorch](https://github.com/huggingface/transformers/tree/main/examples/pytorch), [JAX](https://jax.readthedocs.io/en/latest/), and [PaddlePaddle](https://github.com/PaddlePaddle/Paddle). 
+Additionally, we offer [Jupyter notebook tutorials](https://github.com/NVIDIA/TransformerEngine/tree/main/docs/examples) and a selection of [third-party examples](). Please be aware that these third-party examples might need specific, older versions of dependencies to function properly.
```
Comment (ksivaman) [inline examples/README.md:4]: Not sure if third-party examples needs a link here

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+Please discuss in an [issue](https://github.com/NVIDIA/TransformerEngine/issues) a feature you would like to implement in an example before submitting a PR.
```
Comment (ksivaman) [inline examples/README.md:6]: I think we can generally welcome example PRs without opening an issue first, right?

File: examples/README.md (hunk 3 above)
Comment (sbhavani) [reply examples/README.md:4]: I was going to create a section, let me add that now

File: examples/README.md (hunk 2 above)
Comment (sbhavani) [reply examples/README.md:3]: I'll stick to github repos

**Review by ksivaman — APPROVED** [review]
LGTM, thanks

**Pull request merged**